tests/components/recorder/test_purge.py

```diff
@@ -72,12 +72,15 @@ def test_purge_old_recorder_runs(hass, hass_recorder):
         assert recorder_runs.count() == 7
 
         # run purge_old_data()
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 0, repack=False)
+        assert not finished
+
         finished = purge_old_data(hass.data[DATA_INSTANCE], 0, repack=False)
         assert finished
         assert recorder_runs.count() == 1
```

Comment:
Interestingly, when I run pytest locally, I get an Assertion Error here: `3 == 1`.
```py
pytest tests/components/recorder/test_purge.py
```

Reply:
I've not been able to replicate this on all my test machines or the CI.

Reply:
Tried it locally on my mac as well and the test is passing.

Reply:
Found the issue. My timezone is currently UTC+1, the `purge_old_data` uses `dt_util.utcnow()` for the comparison whereas the tests use `datetime.now()`.

I believe the test should always use `datetime.utcnow()`, but shouldn't `purge_old_data` be using the current user timezone?

Reply:
Edit: Maybe the test should use `dt_util.now(hass.config.time_zone)` as well.

Reply:
Good catch. I’ll have to dig into that later as I’m out of time until later tonight

Reply:
Edit 2: Which is nothing more than `datetime.now()` in the first place 🤦🏻

Reply:
I switch the tests to match the code and use `utcnow`

Reply:
It works now. The confusion with `now` and `utcnow` was my fault. I had falsely assumed that the times in the db are the local once. However, since they use the `UTC` time, `dt_util.utcnow()` is correct.